sound/soc/sof/loader.c

```diff
@@ -116,14 +116,23 @@ int snd_sof_run_firmware(struct snd_sof_dev *sdev)
 	sdev->dbg_dump_printed = false;
 	sdev->ipc_dump_printed = false;
 
-	/* create read-only fw_version debugfs to store boot version info */
 	if (sdev->first_boot) {
+		/* create read-only fw_version debugfs to store boot version info */
 		ret = snd_sof_debugfs_buf_item(sdev, &sdev->fw_version,
 					       sizeof(sdev->fw_version),
 					       "fw_version", 0444);
 		/* errors are only due to memory allocation, not debugfs */
 		if (ret < 0) {
-			dev_err(sdev->dev, "error: snd_sof_debugfs_buf_item failed\n");
+			dev_err(sdev->dev, "error: failed to add fw_version debugfs\n");
+			return ret;
+		}
+
+		ret = snd_sof_debugfs_buf_item(sdev, &sdev->fw_state,
+					       sizeof(sdev->fw_state),
+					       "fw_state", 0444);
```

Comment:
Would not be better to do this in `snd_sof_dbg_init()`? the fw_state is valid before the run_firmware after all...

Reply:
Good point, I'll move it, thanks